portal/static/js/src/gil.js

```diff
@@ -1303,9 +1333,16 @@ module.exports = utilObj = (function() {
     this.hasValue = function(val) {
       return val != null && val != "" && val != "undefined";
     };
-    //this test for full URL - "https://stg-sm.us.truenth.org" etc.
+  
     this.validateUrl = function(val) {
-        return  val && $.trim(val) !== "#" && /^(https?|ftp)?(:)?(\/\/)?([a-zA-Z0-9.-]+(:[a-zA-Z0-9.&%$-]+)*@)*((25[0-5]|2[0-4][0-9]|1[0-9]{2}|[1-9][0-9]?)(\.(25[0-5]|2[0-4][0-9]|1[0-9]{2}|[1-9]?[0-9])){3}|([a-zA-Z0-9-]+\.)*[a-zA-Z0-9-]+\.(com|edu|gov|int|mil|net|org|biz|arpa|info|name|pro|aero|coop|museum|[a-zA-Z]{2}))(:[0-9]+)*(\/($|[a-zA-Z0-9.,?'\\+&%$#=~_-]+))*$/.test(val);
+      if (!val) {
+        return false;
+      }  
+      //this test for full URL - "https://stg-sm.us.truenth.org" etc.
+      if (val.indexOf('://') > 0 || val.indexOf('//') === 0) {
+        return  $.trim(val) !== "#" && /^(https?|ftp)?(:)?(\/\/)?([a-zA-Z0-9.-]+(:[a-zA-Z0-9.&%$-]+)*@)*((25[0-5]|2[0-4][0-9]|1[0-9]{2}|[1-9][0-9]?)(\.(25[0-5]|2[0-4][0-9]|1[0-9]{2}|[1-9]?[0-9])){3}|([a-zA-Z0-9-]+\.)*[a-zA-Z0-9-]+\.(com|edu|gov|int|mil|net|org|biz|arpa|info|name|pro|aero|coop|museum|[a-zA-Z]{2}))(:[0-9]+)*(\/($|[a-zA-Z0-9.,?'\\+&%$#=~_-]+))*$/.test(val);
```

Comment:
I think this is just a test for a generic link (not specific to the SM link). I think the portal should always return a link with the protocol; would you just be able to test for the string `http` or `https`